test/IDE/complete_stmt_controlling_expr.swift

```diff
@@ -635,3 +643,11 @@ func testGuardLetBinding8(_ x: FooStruct?) {
 // FOOSTRUCT_LOCALVAL: Begin completions
 // FOOSTRUCT_LOCALVAL-DAG: Decl[LocalVar]/Local{{(/TypeRelation\[Convertible\])?}}: boundVal[#FooStruct#];
 // FOOSTRUCT_LOCALVAL: End completions
+
+// OPTIONAL_FOOSTRUCT: Begin completions, 5 items
+// OPTIONAL_FOOSTRUCT: Decl[Constructor]/CurrNominal/TypeRelation[Convertible]: init()[#FooStruct#]; name=init()
+// OPTIONAL_FOOSTRUCT: Decl[Constructor]/CurrNominal/TypeRelation[Convertible]: init({#Int#})[#FooStruct#]; name=init(Int)
```

Comment:
How about we add a FIXME to make this `FooStruct` member leakage issue noticeable?

Reply:
Thank you for notifying, will do! Somehow, I thought using optional element member is accepted in matching pattern too. I was wrong.